cmd/client/client.go

```diff
@@ -0,0 +1,94 @@
+package main
```

Comment:
cmd directory should include only main.go to run the program
U should split the project to packages.
I would split the code to cmd, framework and impl (and many packages inside :) )

Reply:
should be something like:
cmd 
    server 
        main.go
    client
        main.go
impl
    server
        communicating
        etc
    client
        communicating
        etc
framework
    messaging
        message.go
        messages
            text.go